### PR TITLE
feat(vcd)(post-ga): add reset VCD password

### PR DESCRIPTION
**Pull request by pauldkn**
ref: MANAGER-15191

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/vcd-postga`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | feat/MANAGER-15191
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/hpc-vmware-managed-vcd/src/data/hooks/useResetPassword.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export const useResetPassword = ({
+  id,
+  onSuccess,
+  onError,
+}: {
+  id: string;
+  onSuccess?: () => void;
+  onError?: (result: ApiError) => void;
+}) => {
```
Comment (tibs245) [inline packages/manager/apps/hpc-vmware-managed-vcd/src/data/hooks/useResetPassword.ts:14]:
```suggestion
}: options: Partial<
    UseMutationOptions<
      YourTypeReturnedOnSuccess,
      Error,
      YourParamsFunction
    >) => {
```

Comment (tibs245) [reply packages/manager/apps/hpc-vmware-managed-vcd/src/data/hooks/useResetPassword.ts:14]:
```suggestion
}: {id: string} & options: Partial<
    UseMutationOptions<
      YourTypeReturnedOnSuccess,
      Error,
      YourParamsFunction
    >) => {
```

File: packages/manager/apps/hpc-vmware-managed-vcd/src/data/hooks/useResetPassword.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { useMutation } from '@tanstack/react-query';
+import { ApiError } from '@ovh-ux/manager-core-api';
+import { resetOrganizationPassword } from '../api/hpc-vmware-managed-vcd-reset-password';
+import { getVcdOrganizationResetPasswordQueryKey } from '@/utils/queryKeys';
+
+export const useResetPassword = ({
+  id,
+  onSuccess,
+  onError,
+}: {
```
Comment (tibs245) [inline packages/manager/apps/hpc-vmware-managed-vcd/src/data/hooks/useResetPassword.ts:10]:
```suggestion
export const useResetPassword = ({
  id,
  ...options
}: {
```

File: packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/OrganizationGeneralInformation.spec.tsx [hunk 3]
```diff
@@ -117,4 +100,66 @@ describe('Organization General Information Page', () => {
...
+    await checkModalVisibility({ container, isVisible: true });
+
+    let validateButton;
+    await waitFor(() => {
+      validateButton = screen.getByText(
+        labels.dashboard.managed_vcd_dashboard_edit_modal_cta_validate,
+      );
+      return expect(validateButton).toBeEnabled();
+    });
+    await waitFor(() => userEvents.click(validateButton));
```
Comment (tibs245) [inline packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/OrganizationGeneralInformation.spec.tsx:134]: I it's not I mistake of me : 

the second waitFor is not use on good use case;

- We wait for text is show : `labels.dashboard.managed_vcd_dashboard_edit_modal_cta_validate`

- And after we wait just the time of click :

await act(userEvents.click(validateButton));

But It's maybe to replace act ?

File: packages/manager/apps/hpc-vmware-managed-vcd/src/components/modal/ResetPasswordModal.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
```
Comment (tibs245) [inline packages/manager/apps/hpc-vmware-managed-vcd/src/components/modal/ResetPasswordModal.tsx:1]: It's not wrong. But I think It's good to think to use React-hook-form for all form for the next conception. So I approve but If you want refactor I am here.

File: packages/manager/apps/hpc-vmware-managed-vcd/src/components/modal/ResetPasswordModal.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      closeModal();
+    },
+  });
+
+  const handleSubmit = async () => {
+    setIsErrorVisible(false);
+    try {
+      await resetPassword();
+    } catch (err) {
+      setIsErrorVisible(true);
```
Comment (chipp972) [inline packages/manager/apps/hpc-vmware-managed-vcd/src/components/modal/ResetPasswordModal.tsx:43]: why not put this in the `onError` callback of `useResetPassword` hook ?

File: packages/manager/apps/hpc-vmware-managed-vcd/src/components/modal/ResetPasswordModal.tsx (hunk 4 above)
Comment (tibs245) [reply packages/manager/apps/hpc-vmware-managed-vcd/src/components/modal/ResetPasswordModal.tsx:1]: Ah ok you have not field , sorry 😅

**Review by tibs245 — APPROVED** [review]

File: packages/manager/apps/hpc-vmware-managed-vcd/src/data/hooks/useResetPassword.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  onError?: (result: ApiError) => void;
+}) => {
+  const { mutateAsync: resetPassword, error, isError } = useMutation({
+    mutationKey: getVcdOrganizationResetPasswordQueryKey(id),
+    mutationFn: () => resetOrganizationPassword(id),
+    onSuccess: () => onSuccess?.(),
+    onError: (result: ApiError) => onError?.(result),
+  });
+
+  return { resetPassword, error, isError };
```
Comment (chipp972) [inline packages/manager/apps/hpc-vmware-managed-vcd/src/data/hooks/useResetPassword.ts:22]: You may need is loading to display a loader while the call is being processed

File: packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/OrganizationGeneralInformation.spec.tsx (hunk 3 above)
Comment (chipp972) [reply packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/OrganizationGeneralInformation.spec.tsx:134]: act is deprecated

File: packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/OrganizationGeneralInformation.spec.tsx [hunk 7]
```diff
@@ -117,4 +100,66 @@ describe('Organization General Information Page', () => {
...
+
+    await checkModalVisibility({ container, isVisible: true });
+
+    let validateButton;
+    await waitFor(() => {
+      validateButton = screen.getByText(
+        labels.dashboard.managed_vcd_dashboard_edit_modal_cta_validate,
+      );
+      return expect(validateButton).toBeEnabled();
+    });
```
Comment (chipp972) [inline packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/OrganizationGeneralInformation.spec.tsx:156]: You could create a function for this, just like checkTextVisibility

File: packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/edit/EditPassword.page.tsx [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import { ResetPasswordModal } from '@/components/modal/ResetPasswordModal';
+
+export default function EditPassword() {
+  return <ResetPasswordModal />;
+}
```
[the remark this answers is not in the document]
Comment (pauldkn) [reply packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/edit/EditPassword.page.tsx:6]: I thought the modal might be accessed from elsewhere and reused, so I made it a component alongside the other modals. But we can definitely move it here if that feels better

[the remark this answers is not in the document]
Comment (chipp972) [reply packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/edit/EditPassword.page.tsx:6]: I think the component is a bit too specific to be reused here

**Review by chipp972 — APPROVED** [review]

File: packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/edit/EditPassword.page.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (pauldkn) [reply packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/organization/general-information/edit/EditPassword.page.tsx]: Yes maybe, I removed it 👍

**Review by tibs245 — APPROVED** [review]

**Pull request merged**